cis194/week4/daniel-deng/src/Exercise03.hs

```diff
@@ -0,0 +1,15 @@
+module CIS194.Homework04.Exercise03 (xor, map' {- , myFoldl -}) where
+
+xor :: [Bool] -> Bool
+xor = foldr (/=) False
+
+map' :: (a -> b) -> [a] -> [b]
+map' f = foldr ((:) . f) []
+
+{-
+foldr f z [x1, x2, ..., xn] == x1 ‘f‘ (x2 ‘f‘ ... (xn ‘f‘ z)...)
+foldl f z [x1, x2, ..., xn] == (...((z ‘f‘ x1) ‘f‘ x2) ‘f‘...) ‘f‘ xn
+
+-- still thinking about how to implement foldl with foldr
+myFoldl :: (a -> b -> a) -> a -> [b] -> a
```

Comment:
Nested function calls, as in one function calls another function calls another function ..., form a stack. And stack is FILO by nature. In another word, the last function being called returns first. Have this in mind, can we accumulating functions using `foldl` so when we invoke these functions, it will return as if we are folding from the right?

Reply:
> Nested function calls, as in one function calls another function calls another function ..., form a stack. And stack is FILO by nature. In another word, the last function being called returns first. Have this in mind, can we accumulating functions using `foldl` so when we invoke these functions, it will return as if we are folding from the right?

Sounds like we are implementing a runtime of a scripting language 😂